parsers/src/main/scala/uk/gov/ons/addressIndex/parsers/Tokens.scala

```diff
@@ -27,9 +56,8 @@ object Tokens extends CrfTokenable {
     val inputWithoutCounties = removeCounties(upperInput)
 
     val tokens = inputWithoutCounties
-      .replace(" - ", "-")
-      .replace("- ", "-")
-      .replace(" -", "-")
+      .replaceAll("(\\d+) ?- ?(\\d+)", "$1-$2")
```

Comment:
Possible improvement to this regex

`.replaceAll("(\\d+).*?-.*?(\\d+)", "$1-$2")`